src/core/ddsc/src/dds_qos.c

```diff
@@ -565,3 +710,91 @@ bool dds_qget_ignorelocal (const dds_qos_t * __restrict qos, dds_ignorelocal_kin
     *ignore = qos->ignorelocal.value;
   return true;
 }
+
+bool dds_qget_propnames (const dds_qos_t * __restrict qos, uint32_t * n, char *** names)
+{
+  bool props;
+  if (qos == NULL || (n == NULL && names == NULL))
+    return false;
+
+  props = (qos->present & QP_PROPERTY_LIST) && qos->property.value.n > 0;
+
+  if (n != NULL)
+    *n = props ? qos->property.value.n : 0;
+
+  if (names != NULL)
+  {
+    if (!props || qos->property.value.n == 0)
```

Comment:
``qos->property.value.n == 0`` means ``props`` is false anyway, so that term serves no purpose.

Reply:
Fixed